leo/cli/commands/devnode/rest/mod.rs

```diff
@@ -0,0 +1,337 @@
+            .route("/blocks", get(Self::get_blocks))
+            .route("/height/{hash}", get(Self::get_height))
+            .route("/statePath/{commitment}", get(Self::get_state_path_for_commitment))
+            .route("/statePaths", get(Self::get_state_paths_for_commitments))
+            .route("/stateRoot/latest", get(Self::get_state_root_latest))
+            .route("/stateRoot/{height}", get(Self::get_state_root));
+
+        // If the `history` feature is enabled, enable the additional endpoint.
+        #[cfg(feature = "history")]
+        let routes = routes.route("/block/{blockHeight}/history/{mapping}", get(Self::get_history));
+
+        routes
+            // Pass in `Rest` to make things convenient.
+            .with_state(self.clone())
+            // Enable tower-http tracing.
+            .layer(TraceLayer::new_for_http())
+            // Custom logging.
+            .layer(middleware::map_request(log_middleware))
+            // Enable CORS.
+            .layer(cors)
+            // Cap the request body size at 512KiB.
+            .layer(DefaultBodyLimit::max(512 * 1024))
+            .layer(GovernorLayer {
+                config: governor_config.into(),
+            })
+    }
+
+    async fn spawn_server(&mut self, rest_ip: SocketAddr, rest_rps: u32) -> Result<()> {
+        // Log the REST rate limit per IP.
+        debug!("REST rate limit per IP - {rest_rps} RPS");
+
+        // Add the v1 API as default and under "/v1".
+        let default_router = axum::Router::new().nest(
+            &format!("/{}", N::SHORT_NAME),
+            self.build_routes(rest_rps).layer(middleware::map_response(v1_error_middleware)),
+        );
+        let v1_router = axum::Router::new().nest(
+            &format!("/{API_VERSION_V1}/{}", N::SHORT_NAME),
+            self.build_routes(rest_rps).layer(middleware::map_response(v1_error_middleware)),
+        );
+
+        // Add the v2 API under "/v2".
+        let v2_router =
+            axum::Router::new().nest(&format!("/{API_VERSION_V2}/{}", N::SHORT_NAME), self.build_routes(rest_rps));
+
+        // Combine all routes.
+        let router = default_router.merge(v1_router).merge(v2_router);
+
+        let rest_listener =
+            TcpListener::bind(rest_ip).await.with_context(|| "Failed to bind TCP port for REST endpoints")?;
+
+        let handle = tokio::spawn(async move {
+            if let Err(e) = axum::serve(rest_listener, router.into_make_service_with_connect_info::<SocketAddr>()).await
+            {
+                eprintln!("REST server crashed: {e:?}");
+            }
+        });
+
+        self.handles.lock().push(handle);
+        Ok(())
+    }
+}
+
+/// Creates a log message for every HTTP request.
+async fn log_middleware(ConnectInfo(addr): ConnectInfo<SocketAddr>, request: Request<Body>) -> Request<Body> {
+    info!("Received {:?} {:?} from {:?}", request.method(), request.uri(), addr);
+    request
+}
+
+/// Converts errors to the old style for the v1 API.
```

Comment:
Is it worth keeping this around? We could just serve the more informative errors for both.